t/40-openqa-clone-job.t

```diff
@@ -3,25 +3,79 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 use Test::Most;
-use Test::Warnings ':report_warnings';
+use Test::Warnings qw(warning :report_warnings);
 use FindBin;
 use lib "$FindBin::Bin/lib", "$FindBin::Bin/../external/os-autoinst-common/lib";
 use OpenQA::Test::TimeLimit '10';
 use OpenQA::Test::Utils qw(run_cmd test_cmd);
-
+use Test::MockModule;
+$ENV{OPENQA_CONFIG} = '';
 
 sub test_once {
     # Report failure at the callsite instead of the test function
     local $Test::Builder::Level = $Test::Builder::Level + 1;
-    # prevent all network access to stay local
-    test_cmd('OPENQA_CONFIG= unshare -r -n script/openqa-clone-job', @_);
+    test_cmd('script/openqa-clone-job', @_);
 }
 
-test_once '', qr/missing.*help for usage/, 'hint shown for mandatory parameter missing', 255, 'needs parameters';
-test_once '--help', qr/Usage:/, 'help text shown', 0, 'help screen is success';
-test_once '--invalid-arg', qr/Usage:/, 'invalid args also yield help', 1, 'help screen on invalid not success';
-my $args = 'http://openqa.opensuse.org/t1';
-test_once $args, qr|API key/secret for 'localhost' missing|, 'fails without API key/secret', 'non-zero', 'fail';
-test_once "--apikey foo --apisecret bar $args", qr/failed to get job '1'/, 'fails without network', 'non-zero', 'fail';
+require "$FindBin::Bin/../script/openqa-clone-job";
+my $main = Test::MockModule->new('main');
+$main->redefine(usage => 0);
+
+my @apiargs = qw(--apikey foo --apisecret bar);
+
+subtest 'usage' => sub {
+    test_once '--help', qr/Usage:/, 'help text shown', 0, 'help screen is success';
+
+    local @ARGV;
+    throws_ok { main::main() } qr/missing.*help for usage/, 'hint shown for mandatory parameter missing';
+    @ARGV = '--invalid-arg';
+    like warning {
+        throws_ok { main::main() } qr/missing job reference/, 'hit for mandatory parameter';
+    }, qr{Unknown option: invalid-arg}, 'expected warning';
+
+    @ARGV = 'http://openqa.local.foo/t1';
+    throws_ok { main::main() } qr|API key/secret for 'localhost' missing|, 'fails without API key/secret';
+};
+
+subtest errors => sub {
+    local @ARGV = (@apiargs, 'http://oopenqa.local.foo/t1');
```

Comment:
I guess you meant `openqa.local.foo` otherwise ignore

Reply:
It is just a non existant url, but I changed it now